src/Components/PaperEditor/ProgrammeScript/index.js

```diff
@@ -1,238 +1,270 @@
+    } else {
+      console.log('Not deleting');
+    }
   };
 
-  handleEdit = i => {
-    const { programmeScript } = this.state;
-    const elements = programmeScript.elements;
-    const currentElement = elements[i];
+  const handleEdit = i => {
+    console.log('Handling edit...');
+    const tempElements = JSON.parse(JSON.stringify(elements));
+    const currentElement = tempElements[i];
     const newText = prompt('Edit', currentElement.text);
     if (newText) {
+      console.log('Editing...');
       currentElement.text = newText;
-      elements[i] = currentElement;
-      programmeScript.elements = elements;
-      // TODO: save to server
-      this.setState({
-        programmeScript: programmeScript,
-        resetPreview: true
-      });
-      // TODO: consider using set state function to avoid race condition? if needed?
-      // this.setState(({ programmeScript }) => {
-      //   return {
-      //     programmeScript: programmeScript
-      //   };
-      // });
+      tempElements[i] = currentElement;
+      setElements(tempElements);
+      setResetPreview(true);
+      console.log('Edited');
+    } else {
+      // either newText is empty or they hit cancel
+      console.log('Not editing');
     }
   };
 
-  handleAddTranscriptElementToProgrammeScript = elementType => {
-    const { programmeScript } = this.state;
-    const elements = this.state.programmeScript.elements;
-    // TODO: refactor - with helper functions
-    if (
-      elementType === 'title' ||
-      elementType === 'note' ||
-      elementType === 'voice-over'
-    ) {
-      const text = prompt(
-        'Add some text for a section title',
-        'Some place holder text'
-      );
-
-      const indexOfInsertPoint = this.getIndexPositionOfInsertPoint();
-      const newElement = {
-        id: cuid(),
-        index: elements.length,
-        type: elementType,
-        text: text
-      };
-      elements.splice(indexOfInsertPoint, 0, newElement);
-      programmeScript.elements = elements;
-      // TODO: save to server
-      this.setState({
-        programmeScript: programmeScript,
-        resetPreview: true
-      });
-    }
+  const onSortEnd = ({ oldIndex, newIndex }) => {
+    const result = arrayMove(elements, oldIndex, newIndex);
+    handleReorder(result);
+    setElements(result);
   };
 
-  getIndexPositionOfInsertPoint = () => {
-    const { programmeScript } = this.state;
-    const elements = programmeScript.elements;
-    // find insert point in list,
+  const getIndexPositionOfInsertPoint = () => {
     const insertPointElement = elements.find(el => {
       return el.type === 'insert';
     });
-    // get insertpoint index
     const indexOfInsertPoint = elements.indexOf(insertPointElement);
 
     return indexOfInsertPoint;
   };
 
-  // TODO: save to server
-  // TODO: needs to handle when selection spans across multiple paragraphs
-  handleAddTranscriptSelectionToProgrammeScript = () => {
+  // // TODO: save to server
```

Comment:
Deciding when to save automatically is out of scope of this ticket. I'll make a note - might be a new issue.